tests/test_lstm.py

```diff
@@ -54,49 +55,149 @@ def future_loading2(t, x = None):
+                    'dt': TIMESTEP
+                })
+            x_counter = m.next_state(x_counter, future_loading(t), t - t_counter)
+            t_counter = t
+            return z
+        
+        results2 = m3.simulate_to(7.5, future_loading2, dt=TIMESTEP, save_freq=TIMESTEP)
+
+        # Have to do it this way because the other way (i.e., using the LSTM model- the states are not a subset)
+        # Compare RMSE of the results to the original data
+        error = m.calc_error(results2.times, results2.inputs, results2.outputs)
+        self.assertLess(error, 2)
+
+        # Now with no 'dt'
+        #----------------------------------------------------
+        m3 = LSTMStateTransitionModel.from_model(
+            m,
+            [future_loading for _ in range(5)],
+            dt = [TIMESTEP, TIMESTEP/2, TIMESTEP/4, TIMESTEP*2, TIMESTEP*4],
+            window=2, 
+            epochs=30,
+            add_dt = False)  
+
+        # Should get keys from original model
+        self.assertSetEqual(set(m3.inputs), set(['x_t-1']))
+        self.assertSetEqual(set(m3.outputs), set(m.outputs))
+
+         # Step 3: Use model to simulate_to time of threshold
+        t_counter = 0
+        x_counter = m.initialize()
+        def future_loading2(t, x = None):
+            # Future Loading is a bit complicated here 
+            # Loading for the resulting model includes the data inputs, 
+            # and the output from the last timestep
+            nonlocal t_counter, x_counter
+            z = m.output(x_counter)
+            z = m3.InputContainer(
+                {
+                    'x_t-1': z['x']
+                })
+            x_counter = m.next_state(x_counter, future_loading(t), t - t_counter)
+            t_counter = t
+            return z
+        
+        results2 = m3.simulate_to(7.5, future_loading2, dt=TIMESTEP, save_freq=TIMESTEP)
+
+        # Have to do it this way because the other way (i.e., using the LSTM model- the states are not a subset)
+        # Compare RMSE of the results to the original data
+        error = m.calc_error(results2.times, results2.inputs, results2.outputs)
+        self.assertLess(error, 2)
+
     def test_improper_input(self):
-        with self.assertRaises(ValueError, msg='No inputs'):
-            LSTMStateTransitionModel.from_data([])
-        with self.assertRaises(ValueError, msg="Not iterable"):
-            LSTMStateTransitionModel.from_data(12)
-        with self.assertRaises(ValueError, msg="Empty Tuple"):
-            LSTMStateTransitionModel.from_data(())
-        with self.assertRaises(ValueError, msg="Small Tuple"):
-            LSTMStateTransitionModel.from_data((1))
-        with self.assertRaises(ValueError, msg="Too large Tuple"):
-            LSTMStateTransitionModel.from_data((1, 2, 3))
+        # Input, output format
+        with self.assertRaises(Exception, msg='No outputs'):
+            LSTMStateTransitionModel.from_data(inputs=[])
```

Comment:
Is this line flipped with line 149? I'm not sure I completely understand the syntax of the testing here, but I think lines 146-147 are testing what happens when "inputs = []" and this raises the exception "No outputs" ... should these be the same (either inputs or outputs, with the other tested in 148-149)

Reply:
Actually this is right. You're right that line 147 corresponds with the check on 146. In line 147 inputs are provided (an empty array) but no outputs at all. I can see how this would be confusing since I'm essentially providing no inputs since the array is empty. I think it will cause less confusion if I put a number in there. 

```suggestion
            LSTMStateTransitionModel.from_data(inputs=[1])
```